### ISSUE TITLE
Load function broken in HTML5 export.

### ISSUE THREAD

**Issue by That1Guard** [That1Guard/UndertaleDemoGMS#1]
**Problem:** 
Load function seems to break when trying to load an original save from it, that would otherwise would be fine for use in desktop compilation of the game.

**Stacktrace:**
Error: Error: Room 2507 is not a valid room index
--------------------------------------------------------------------
	function _Wt("Error: Room 2507 is not a valid room index")
	function room_goto(2507)
	function gml_Script_scr_load([instance], [instance])
	function gml_Script_scr_namingscreen([instance], [instance])
	function gml_Object_obj_intromenu_Draw_0([instance], [instance])
	function(2048, 0, [instance], [instance])
	function(2048, 0, [instance], [instance])
	function([unknown])
	function([unknown])
	function([unknown])
	function()
	function _QN2()
	function _sN2()
	function _pN2(11184.73)

**Relevant Files:**

- **file9:**
`A
20
20
10
0
10
0
4
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
3
4
0
0
0
0
0
84
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
9
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
14
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
2
1
1
0
233
6
2507
NaN`

- **undertale.ini**
`[General]
fun="84"
Name="A"
Love="1"
Time="2507"
Kills="0"
Room="6"
[Flowey]
Met1="1"
`

- **file0**
`A
1
20
20
10
0
10
0
4
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
3
4
0
0
0
0
0
84
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
9
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
14
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
2
1
1
0
233
6
2507`

**Comment by ethanaobrien** [comment]
@That1Guard I can possibly fix this - I have experience with javascript it is most likely a problem with the file being truncated or not saved properly.

Wasnt html5 added in gamemaker 2?

**Comment by That1Guard** [comment]
> @That1Guard I can possibly fix this - I have experience with javascript it is most likely a problem with the file being truncated or not saved properly.
> 
> Wasnt html5 added in gamemaker 2?

Correct, there is an HTML5 module for GameMaker Studio 2. Though this was originally built with the GMS 1.4, I'm unsure of what differences you might find with GMS 1.4 and GMS 2 if intend to use that (though the demo itself was built with GM8, so... maybe none).

**Comment by ethanaobrien** [comment]
@That1Guard I just did some looking and confirmed, it is because the file it too big to be stored in localstorage. This would have to be fixed in the javascript side. Would you like me to do this? It would be relatively easy to fix

**Comment by That1Guard** [comment]
> @That1Guard I just did some looking and confirmed, it is because the file it too big to be stored in localstorage. This would have to be fixed in the javascript side. Would you like me to do this? It would be relatively easy to fix

Sure, go ahead if you want.

**Comment by ethanaobrien** [comment]
@That1Guard
Is it possible to export a gamemaker game as `C` or `C++`? If I had these I could compile it to webassembly.
If its possible could you send me the exported `C`/`C++` files as I can't get the gamemaker software myself

**Comment by That1Guard** [comment]
> @That1Guard
> Is it possible to export a gamemaker game as `C` or `C++`? If I had these I could compile it to webassembly.
> If its possible could you send me the exported `C`/`C++` files as I can't get the gamemaker software myself

There is currently no way to transpile GML code into C/C++ at the current time since direct compilation into C/C++ is when they get baked by obfuscation into the application.

If you need help with retrieval and activation of GMS 1.4, then you can send me a DM to Wendy's#0666, and I can provide information there should you ask.

**Comment by ethanaobrien** [comment]
Thank you, I will do so when I get the chance

**Comment by ethanaobrien** [comment]
@That1Guard I sent you a friend request. It said I cant send you messages until it is approved

**Comment by That1Guard** [comment]
@ethanaobrien Accepted FR.
